pkg/storage/store_pool_test.go

```diff
@@ -318,6 +321,101 @@ func TestStorePoolGetStoreList(t *testing.T) {
 	}
 }
 
+func TestStorePoolUpdateLocalStore(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	manual := hlc.NewManualClock(123)
+	clock := hlc.NewClock(manual.UnixNano, time.Nanosecond)
+	// We're going to manually mark stores dead in this test.
+	stopper, g, _, sp, _ := createTestStorePool(
+		TestTimeUntilStoreDead, false /* deterministic */, nodeStatusDead)
+	defer stopper.Stop(context.TODO())
+	sg := gossiputil.NewStoreGossiper(g)
+	stores := []*roachpb.StoreDescriptor{
+		{
+			StoreID: 1,
+			Node:    roachpb.NodeDescriptor{NodeID: 1},
+			Capacity: roachpb.StoreCapacity{
+				Capacity:        100,
+				Available:       50,
+				RangeCount:      5,
+				LogicalBytes:    30,
+				WritesPerSecond: 30,
+			},
+		},
+		{
+			StoreID: 2,
+			Node:    roachpb.NodeDescriptor{NodeID: 2},
+			Capacity: roachpb.StoreCapacity{
+				Capacity:        100,
+				Available:       55,
+				RangeCount:      4,
+				LogicalBytes:    25,
+				WritesPerSecond: 25,
+			},
+		},
+		{
+			StoreID: 3,
+			Node:    roachpb.NodeDescriptor{NodeID: 3},
+			Capacity: roachpb.StoreCapacity{
+				Capacity:        100,
+				Available:       45,
+				RangeCount:      6,
+				LogicalBytes:    35,
+				WritesPerSecond: 35,
+			},
+		},
+	}
+	sg.GossipStores(stores, t)
+	node := roachpb.NodeDescriptor{NodeID: roachpb.NodeID(1)}
+	eng := engine.NewInMem(roachpb.Attributes{}, 1<<20)
+	stopper.AddCloser(eng)
+	cfg := TestStoreConfig(clock)
+	cfg.Transport = NewDummyRaftTransport(cfg.Settings)
+	store := NewStore(cfg, eng, &node)
```

Comment:
It seems like some of this setup could be skipped (e.g. it doesn't look like we use store 3 for anything; do we really need to create a real engine and store for the replica that never has to actually do anything?).

Reply:
I have removed store 3.
If we  didn't create a real engine and store, when we want to create a replica with `NewReplica()`, it will always coredump.